### ISSUE TITLE
Option to disable specific ligatures globally

### ISSUE THREAD

**Issue by archetyped** [CoenraadS/vscode-Disable-Ligatures#2]
Many ligatures are useful, but some can be difficult to differentiate at a glance in certain fonts, resulting in more time spent evaluating code.  As such, I would love the option to disable specific ligatures for the entire editor view (i.e. not at just the selected line or cursor position).

Example:
```json
"disableLigatures.ligatures.global": [
  "==",
  "===",
  "!=",
  "!==",
]
```

Thanks!

**Comment by CoenraadS** [comment]
Doesn't `"editor.fontLigatures": false` already do that?

**Comment by archetyped** [comment]
`"editor.fontLigatures": false` disables support for **all** ligatures. It does not allow you to specify **specific** ligatures to disable globally as per the example in my original post.

**Comment by CoenraadS** [comment]
ah I understand now

**Comment by kennethreitz** [comment]
+1000 for this!

**Comment by jonathantneal** [comment]
I wanted to this too! I’ve made a pull request to add this functionality. Perhaps someone could help by reviewing the changes and easing @CoenraadS’s time?

https://github.com/CoenraadS/vscode-Disable-Ligatures/pull/3

**Comment by mbeiswenger** [comment]
@CoenraadS Any intention on pulling this PR or is this repo abandoned?

**Comment by CoenraadS** [comment]
I will take a look at it tonight. Thanks

**Comment by cgbeutler** [comment]
Ligatures are natively supported. I requested the feature on VSCode itself.
UP-VOTES NEEDED to make it happen.
https://github.com/microsoft/vscode/issues/98758#event-3385790303

**Comment by mparsakia** [comment]
Agreed, ligature for '===' can look really bad on some of my displays, I would like to be able to disable it.

**Comment by cgbeutler** [comment]
@mparsakia  Yeah, sadly it didn't get enough upvotes in time. Apparently we are too few.
There may be a way to write a plugin for this? Scan for ligatures and undo them? Don't know.

**Comment by mparsakia** [comment]
@cgbeutler Wow, I just realized this is actually the repo of the extension that does this; search in VSCode for "Disable Ligatures" by CoenraadS, then go to your 'settings.json' and you can set   `"disableLigatures.ligatures": ["===", ...]` for each type of ligature you would like to disable. It seems like it will disable it everywhere except the settings.json file for now, but I may need to restart.

**Comment by cgbeutler** [comment]
@mparsakia Thought I tried that feature and it didn't work. Maybe I didn't restart....

**Comment by cgbeutler** [comment]
As a side note, if you are using FiraCode-based fonts, you can sometimes give it optional font ligature settings found here: https://github.com/tonsky/FiraCode/wiki/How-to-enable-stylistic-sets

Unfortunately, those settings are not share by all ligature fonts.

**Comment by ilyagr** [comment]
> @cgbeutler Wow, I just realized this is actually the repo of the extension that does this; search in VSCode for "Disable Ligatures" by CoenraadS, then go to your 'settings.json' and you can set `"disableLigatures.ligatures": ["===", ...]` for each type of ligature you would like to disable. It seems like it will disable it everywhere except the settings.json file for now, but I may need to restart.

I think that setting has a misleading name, and it doesn't do what I hoped it would. For me, it only applies under the cursor (or over the line the cursor is on, depending on the value of `disableLigatures.mode`). Ligatures *not* in `disableLigatures.ligatures` remain enabled even when the cursor is over them.

**Comment by alfonsusac** [comment]
The `"disableLigatures.ligatures": ["===", ...]` does nothing on my workspace, even after restarting :/
